src/main/java/wolox/training/models/Book.java

```diff
@@ -0,0 +1,128 @@
+package wolox.training.models;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity	
+class Book {
+  
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;	
+ 
+    @Column(nullable = false)
+    private String author;
+    
+    @Column(nullable = false)
+    private String image;
+    
+    @Column(nullable = false, unique = true)
+    private String title;
+    
+    @Column(nullable = false)
+    private String subtitle;
+    
+    @Column(nullable = false) 
+    private String publisher;
+    
+    @Column(nullable = false)
+    private String year;
+    
+    @Column(nullable = false)
+    private int pages;
+    
+    @Column(nullable = false)
+    private String isbn;

```

Comment:
Estos campos que sí son requeridos podrían usar la annotation @NotNull (de javax.validation.constraints.NotNull) que les comenté. La diferencia está en que con @NotNull JPA entra en acción atajando nulls antes de que lleguen a la base de datos (Con @column(nullable = false) la validación directamente se haría en la base de datos). Si bien en el link de la card se utiliza @Column, vamos a preferir @NotNull para estos casos así evitamos el chequeo en la DB 👍

Reply:
No me toma el import que mencionas, me sugiere el com.sun.istack.NotNull

Reply:
Mmm interesante jaja cuando puedas hacemos una call si te parece. Pero debería tomarlo 🤔 Incluso tendría que recomendártelo al querer usar `@NotNull`. Por ejemplo acá se puede ver en la clase de Jony:

https://github.com/wolox-training/js-java/blob/feature-modelo-de-book/src/main/java/wolox/training/models/Book.java

Reply:
> 
> 
> Mmm interesante jaja cuando puedas hacemos una call si te parece. Pero debería tomarlo 🤔 Incluso tendría que recomendártelo al querer usar `@NotNull`. Por ejemplo acá se puede ver en la clase de Jony:
> 
> https://github.com/wolox-training/js-java/blob/feature-modelo-de-book/src/main/java/wolox/training/models/Book.java

Ahi te muestro en un pantallazo!

me dice "The import javax.validation cannot be resolved"

<img width="960" alt="bookentity" src="https://user-images.githubusercontent.com/45366646/80540488-d542d400-897f-11ea-9ed3-3a4be56369ac.png">

Reply:
Ahí encontré el problema! En tu pom.xml estás usando una versión de Spring Boot que no es una release:
```
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.3.0.M4</version>
		<relativePath/> <!-- lookup parent from repository -->
	</parent>
```

Habría que cambiar la versión por `2.2.6.RELEASE`

Reply:
Dejemos el pom.xml así directamente:

```

<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.2.6.RELEASE</version>
		<relativePath/> <!-- lookup parent from repository -->
	</parent>
	<groupId>wolox</groupId>
	<artifactId>training</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<name>training</name>
	<description>Demo project for Spring Boot</description>

	<properties>
		<java.version>1.8</java.version>
	</properties>

	<dependencies>
		<dependency>
                     <groupId>org.springframework.boot</groupId>
                     <artifactId>spring-boot-starter-thymeleaf</artifactId>
		</dependency>
		<dependency>
                     <groupId>org.springframework.boot</groupId>
                     <artifactId>spring-boot-starter-web</artifactId>
		</dependency>
		<dependency>
                     <groupId>org.springframework.boot</groupId>
                     <artifactId>spring-boot-starter-data-jpa</artifactId>
		</dependency>
		<dependency>
                     <groupId>com.h2database</groupId>
                     <artifactId>h2</artifactId>
                     <version>1.4.199</version>
		</dependency>
		<dependency>
                     <groupId>org.springframework.boot</groupId>
                     <artifactId>spring-boot-devtools</artifactId>
                     <scope>runtime</scope>
                     <optional>true</optional>
		</dependency>
		<dependency>
                     <groupId>org.springframework.boot</groupId>
                     <artifactId>spring-boot-starter-test</artifactId>
                     <scope>test</scope>
                     <exclusions>
                         <exclusion>
                             <groupId>org.junit.vintage</groupId>
                                 <artifactId>junit-vintage-engine</artifactId>
                         </exclusion>
                     </exclusions>
		</dependency>
	</dependencies>

	<build>
            <plugins>
                 <plugin>
                     <groupId>org.springframework.boot</groupId>
                     <artifactId>spring-boot-maven-plugin</artifactId>
                 </plugin>
             </plugins>
        </build>

</project>

```

Reply:
sospechaba que venia por ese lado. Pero no lo iba a encontrar nunca... Genial!

Ahi te hago un push

Reply:
Groso! 😄